executor/importer/table_import.go

```diff
@@ -58,32 +58,45 @@ var (
 	CheckDiskQuotaInterval = time.Minute
 )
 
-func prepareSortDir(e *LoadDataController, taskID int64) (string, error) {
-	tidbCfg := tidb.GetGlobalConfig()
+// prepareSortDir creates a new directory for import, remove previous sort directory if exists.
+func prepareSortDir(e *LoadDataController, taskID int64, tidbCfg *tidb.Config) (string, error) {
 	sortPathSuffix := "import-" + strconv.Itoa(int(tidbCfg.Port))
-	sortPath := filepath.Join(tidbCfg.TempDir, sortPathSuffix, strconv.FormatInt(taskID, 10))
+	importDir := filepath.Join(tidbCfg.TempDir, sortPathSuffix)
+	sortDir := filepath.Join(importDir, strconv.FormatInt(taskID, 10))
 
-	if info, err := os.Stat(sortPath); err != nil {
+	info, err := os.Stat(importDir)
+	switch {
+	case err != nil && !os.IsNotExist(err):
+		e.logger.Error("stat import dir failed", zap.String("import_dir", importDir), zap.Error(err))
+		return "", errors.Trace(err)
+	case os.IsNotExist(err) || !info.IsDir():
 		if !os.IsNotExist(err) {
-			e.logger.Error("stat sort dir failed", zap.String("path", sortPath), zap.Error(err))
-			return "", errors.Trace(err)
+			e.logger.Warn("import dir is not a dir, remove it", zap.String("import_dir", importDir))
+			if err := os.RemoveAll(importDir); err != nil {
+				return "", errors.Trace(err)
+			}
 		}
-	} else if info.IsDir() {
-		// Currently remove all dir to clean garbage data.
-		// TODO: when do checkpoint should change follow logic.
-		err := os.RemoveAll(sortPath)
-		if err != nil {
-			e.logger.Error("remove sort dir failed", zap.String("path", sortPath), zap.Error(err))
+		e.logger.Info("import dir not exists, create it", zap.String("import_dir", importDir))
+		if err := os.MkdirAll(importDir, 0o700); err != nil {
+			e.logger.Error("failed to make dir", zap.String("import_dir", importDir), zap.Error(err))
+			return "", errors.Trace(err)
 		}
+	default: // err==nil && info.IsDir()
 	}
 
-	err := os.MkdirAll(sortPath, 0o700)
-	if err != nil {
-		e.logger.Error("failed to make sort dir", zap.String("path", sortPath), zap.Error(err))
+	// todo: remove this after we support checkpoint
+	_, err = os.Stat(sortDir)
+	switch {
+	case err != nil && !os.IsNotExist(err):
+		e.logger.Error("stat sort dir failed", zap.String("sort_dir", sortDir), zap.Error(err))
 		return "", errors.Trace(err)
+	case !os.IsNotExist(err):
+		e.logger.Warn("sort dir already exists, remove it", zap.String("sort_dir", sortDir))
+		if err := os.RemoveAll(sortDir); err != nil {
```

Comment:
this will remove sortDir itself, so we only make sure `sortDir` not exists in here?